lms/djangoapps/grades/new/subsection_grade.py

```diff
@@ -114,6 +114,8 @@ def init_from_model(self, student, model, course_structure, submissions_scores,
         Load the subsection grade from the persisted model.
         """
         for block in model.visible_blocks.blocks:
+            if block.locator not in set(course_structure.post_order_traversal(start_node=self.location)):
```

Comment:
This is going to be a very expensive operation for large courses - as this becomes an O(n^2) from an O(n) operation, where n = number of blocks in the course.